__tests__/app.test.js

```diff
@@ -14,20 +14,57 @@ afterAll(() => {
 
 describe("GET /api/topics", () => {
   test("should return with a 200 status code", () => {
-    return request(app).get("/api/topics").expect(200)
+    return request(app).get("/api/topics").expect(200);
   });
 
   test("should return with a property of slug and description", () => {
-    return request(app).get("/api/topics").expect(200).then(({body})=>{body.forEach(({
-        description,
-        slug
-    })=>{
-        expect(typeof description).toBe("string")
-        expect(typeof slug).toBe("string")
-    }
-
-    )
-
-    })
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        body.forEach(({ description, slug }) => {
+          expect(typeof description).toBe("string");
+          expect(typeof slug).toBe("string");
+        });
+      });
+  });
+});
+describe("Get /api", () => {
+  test("should return An object describing all the available endpoints on your API", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((body) => {
+        expect(JSON.parse(body.text)).toEqual({
+          "GET /api": {
+            description:
+              "serves up a json representation of all the available endpoints of the api",
+          },
+          "GET /api/topics": {
+            description: "serves an array of all topics",
+            queries: [],
+            exampleResponse: {
+              topics: [{ slug: "football", description: "Footie!" }],
+            },
+          },
+          "GET /api/articles": {
+            description: "serves an array of all articles",
+            queries: ["author", "topic", "sort_by", "order"],
+            exampleResponse: {
+              articles: [
+                {
+                  title: "Seafood substitutions are increasing",
+                  topic: "cooking",
+                  author: "weegembump",
+                  body: "Text from the article..",
+                  created_at: "2018-05-30T15:59:13.341Z",
+                  votes: 0,
+                  comment_count: 6,
+                },
+              ],
```

Comment:
You will have to update this test after every endpoint. You can require in the json file and check that it equals that instead